### ISSUE TITLE
Upgrade provider version to hashicorp/azurerm 2.49.0

### ISSUE THREAD

**Issue by jan-ehlers** [sassoftware/viya4-iac-azure#131]
We have ran into an issue with v. 2.42 which is required in main.tf. It seems to be fixed with v 2.49. 

There is a discussion about this bug on https://github.com/terraform-providers/terraform-provider-azurerm/issues/10292

The error we see is:

Error: Error making Read request on AzureRM Log Analytics workspaces 'mnr-sas-viya-log-dev': azure.BearerAuthorizer#WithAuthorization: Failed to refresh the Token for request to https://management.azure.com/subscriptions/bdd1ead1-0e94-4637-93a4-acea65b73391/resourcegroups/MarketsAndRisk-SAS-Viya-DEV/providers/Microsoft.OperationalInsights/workspaces/mnr-sas-viya-log-dev?api-version=2020-08-01: StatusCode=400 -- Original Error: adal: Refresh request failed. Status Code = '400'. Response body: {"error":{"code":"bad_request_102","message":"Required metadata header not specified"}} Endpoint http://localhost:50342/oauth2/token?api-version=2017-09-01&resource=https%3A%2F%2Fmanagement.azure.com%2F

**Comment by andybouts** [comment]
I seem to be having the same issue.  I was previously able to build with the IAC, but today, I am unable to get past this error.  The IAC seems to build about 3/4 of the objects, and then fails.  If it really was a problem with my SP, I would expect nothing to build.

```$ tf apply -auto-approve
data.external.iac_tooling_version: Refreshing state... [id=-]
data.external.git_hash: Refreshing state... [id=-]

Error: Error building account: Error getting authenticated object ID: Error listing Service Principals: autorest.DetailedError{Original:adal.tokenRefreshError{message:"adal: Refresh request failed. Status Code = '401'. Response body: {\"error\":\"invalid_client\",\"error_description\":\"AADSTS7000215: Invalid client secret is provided.\\r\\nTrace ID: 8{}100\\r\\nCorrelation ID: db480{}f4881\\r\\nTimestamp: 2021-04-05 17:45:43Z\",\"error_codes\":[7000215],\"timestamp\":\"2021-04-05 17:45:43Z\",\"trace_id\":\"8e5024{}a100\",\"correlation_id\":\"db48{}4881\",\"error_uri\":\"https://login.microsoftonline.com/error?code=7000215\"} Endpoint https://login.microsoftonline.com/b1c1{}a0c2f/oauth2/token?api-version=1.0", resp:(*http.Response)(0xc000cce240)}, PackageType:"azure.BearerAuthorizer", Method:"WithAuthorization", StatusCode:401, Message:"Failed to refresh the Token for request to https://graph.windows.net/b1c14d5c-3625-45b3-a430-9552373a0c2f/servicePrincipals?%24filter=appId+eq+%27806{}2%27&api-version=1.6", ServiceError:[]uint8(nil), Response:(*http.Response)(0xc000cce240)}

  on main.tf line 1, in provider "azurerm":
   1: provider "azurerm" {

**Comment by andybouts** [comment]
Thanks @thpang for pointing me to the credential reset command on this page which fixed the issue: https://github.com/sassoftware/viya4-iac-azure/blob/main/docs/user/AzureHelpTopics.md#how-to-create-service-principal-using-azurecli

**Comment by thpang** [comment]
This was addressed with #137
